openedx/core/djangoapps/programs/rest_api/v1/views.py

```diff
@@ -16,80 +17,74 @@
-                    "width": 726,
-                    "height": 242
-                },
-                "small": {
-                    "url": "http://localhost:18381/media/programs/banner_images/ff41a5eb-2a73-4933-8e80.small.jpg",
-                    "width": 435,
-                    "height": 145
-                },
-                "x-small": {
-                    "url": "http://localhost:18381/media/programs/banner_images/ff41a5eb-2a73-4933-8e8.x-small.jpg",
-                    "width": 348,
-                    "height": 116
-                }
-            },
-            "authoring_organizations": [
+            [
                 {
-                    "key": "edX"
+                    "uuid": "ff41a5eb-2a73-4933-8e80-a1c66068ed2c",
+                    "title": "Demonstration Program",
+                    "type": "MicroMasters",
+                    "banner_image": {
+                        "large": {
+                            "url": "http://example.com/images/foo.large.jpg",
+                            "width": 1440,
+                            "height": 480
+                        },
+                        "medium": {
+                            "url": "http://example.com/images/foo.medium.jpg",
+                            "width": 726,
+                            "height": 242
+                        },
+                        "small": {
+                            "url": "http://example.com/images/foo.small.jpg",
+                            "width": 435,
+                            "height": 145
+                        },
+                        "x-small": {
+                            "url": "http://example.com/images/foo.x-small.jpg",
+                            "width": 348,
+                            "height": 116
+                        }
+                    },
+                    "authoring_organizations": [
+                        {
+                            "key": "example"
+                        }
+                    ],
+                    "progress": {
+                        "uuid": "ff41a5eb-2a73-4933-8e80-a1c66068ed2c",
+                        "completed": 0,
+                        "in_progress": 0,
+                        "not_started": 2
+                    }
                 }
-            ],
-            "progress": {
-                "uuid": "ff41a5eb-2a73-4933-8e80-a1c66068ed2c",
-                "completed": 0,
-                "in_progress": 0,
-                "not_started": 2
-            }
-        }
-    ]
-    """
-
-    permission_classes = (IsAuthenticated,)
-
-    def get(self, request, enterprise_uuid):
-        """
-        Return a list of a enterprise learner's all enrolled programs with their progress.
-
-        Args:
-            request (Request): DRF request object.
-            enterprise_uuid (string): UUID of an enterprise customer.
+            ]
         """
-        user = request.user
+        user: "AnonymousUser | User" = request.user
```

Comment:
technically it will never be `AnonymousUser`  because these are both authenticated endpoints, and also technically we should really be using `get_user_model`  all the time, but again, this is just type checking. This is good enough for that purpose.